File: DynamicProgramming1/RGBStreet/GoRGBStreet/RGBStreet.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

type cost struct {
	r int
	g int
	b int
}

const (
	max = 1000
)

var (
	costList [max + 1]cost
)

func min(a, b int) int {
	if a > b {
		return b
	}
	return a
}

func main() {
	br := bufio.NewReader(os.Stdin)
	bw := bufio.NewWriter(os.Stdout)

	n := 0
	fmt.Fscanln(br, &n)

	for i := 1; i <= n; i++ {
		temp := cost{0, 0, 0}
		fmt.Fscanln(br, &temp.r, &temp.g, &temp.b)

		costList[i].r = temp.r + min(costList[i-1].g, costList[i-1].b)
		costList[i].g = temp.g + min(costList[i-1].r, costList[i-1].b)
		costList[i].b = temp.b + min(costList[i-1].r, costList[i-1].g)
	}

	minCost := min(min(costList[n].r, costList[n].g), costList[n].b)

	fmt.Fprintln(bw, minCost)
	bw.Flush()
}

File: DynamicProgramming1/RGBStreet/CRGBStreet/RGBStreet.c
#include <stdio.h>

#define MAX 1000
#define min(a, b) ((a) < (b) ? (a) : (b))

struct Cost
{
	int R;
	int G;
	int B;
} cost[MAX + 1];

int main()
{
	int n;
	struct Cost temp;
	int min_cost;

	scanf_s("%d", &n);

	for (int i = 1; i <= n; i++)
	{
		scanf_s("%d %d %d", &temp.R, &temp.G, &temp.B);

		cost[i].R = temp.R + min(cost[i - 1].G, cost[i - 1].B);
		cost[i].G = temp.G + min(cost[i - 1].R, cost[i - 1].B);
		cost[i].B = temp.B + min(cost[i - 1].R, cost[i - 1].G);
	}

	min_cost = min(min(cost[n].R, cost[n].G), cost[n].B);

	printf("%d\n", min_cost);

	return 0;
}
